lib/file-store.js

```diff
@@ -333,172 +323,120 @@ const FileStore = function(rootDirectory) {
           metaFile: metaFile,
           headers: req.headers
         },
-        function(err, metaData) {
-          if (err) {
-            return done("Error uploading file");
-          }
-          return done(null, new S3Object(metaData));
+        (err, metaData) => {
+          if (err) return done("Error uploading file");
+          done(null, new S3Object(metaData));
         }
       );
     });
-  };
-
-  const copyObject = function(options, done) {
-    const req = options.request;
-    const srcBucket = options.srcBucket;
-    const srcKey = options.srcKey;
-    const destBucket = options.destBucket;
-    const destKey = options.destKey;
-    const srcKeyPath = path.resolve(getBucketPath(srcBucket.name), srcKey);
-    const destKeyPath = path.resolve(getBucketPath(destBucket.name), destKey);
-    const replaceMetadata = options.replaceMetadata;
-    const srcMetadataFilePath = path.join(srcKeyPath, METADATA_FILE);
+  }
+
+  copyObject(
+    { request, srcBucket, srcKey, destBucket, destKey, replaceMetadata },
+    done
+  ) {
+    const srcKeyPath = path.join(this.getBucketPath(srcBucket.name), srcKey);
+    const destKeyPath = path.join(this.getBucketPath(destBucket.name), destKey);
     const srcContentFilePath = path.join(srcKeyPath, CONTENT_FILE);
-    const destMetadataFilePath = path.join(destKeyPath, METADATA_FILE);
+    const srcMetadataFilePath = path.join(srcKeyPath, METADATA_FILE);
     const destContentFilePath = path.join(destKeyPath, CONTENT_FILE);
+    const destMetadataFilePath = path.join(destKeyPath, METADATA_FILE);
 
     if (srcKeyPath !== destKeyPath) {
       fs.mkdirpSync(destKeyPath);
       fs.copySync(srcContentFilePath, destContentFilePath);
     }
 
     if (replaceMetadata) {
-      const originalObject = buildS3ObjectFromMetaDataFile(
+      const originalObject = this.buildS3ObjectFromMetaDataFile(
         srcKey,
         fs.readFileSync(srcMetadataFilePath)
       );
-      createMetaData(
+      this.createMetaData(
         {
+          key: destKey,
           contentFile: destContentFilePath,
+          metaFile: destMetadataFilePath,
           type: originalObject.contentType,
           encoding: originalObject.contentEncoding,
           disposition: originalObject.contentDisposition,
-          key: destKey,
-          metaFile: destMetadataFilePath,
-          headers: req.headers
+          headers: request.headers
         },
-        function(err, metaData) {
-          if (err) {
-            return done("Error updating metadata");
-          }
-          return done(null, new S3Object(metaData));
+        (err, metaData) => {
+          if (err) return done("Error updating metadata");
+          done(null, new S3Object(metaData));
         }
       );
-    } else {
+    } else if (srcKeyPath !== destKeyPath) {
```

Comment:
This is a relic of me messing around for the fix I made in #131 that I accidentally left in here. `controllers.js` should actually be checking to make sure that the source and destination aren't the same since there's a specific error response S3 will send if you forget to specify the REPLACE MetadataDirective. I'll properly address this in an upcoming PR, but I'll revert this for now.